### PR TITLE
fix: Add missing ios support for balance input

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

File: packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx [hunk 1]
```diff
@@ -9,12 +9,20 @@ const BalanceInput: React.FC<BalanceInputProps> = ({
   onChange,
   currencyValue,
   inputProps,
+  inputMode = "numeric",
```
Comment (RabbitDoge) [inline packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx:12]: No need for a parameter, it can be hardcoded inside the component directly

**Review by RabbitDoge — CHANGES_REQUESTED** [review]

File: packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx (hunk 1 above)
Comment (memoyil) [reply packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx]: I removed inputmode from params, change inputmode default to decimal and add min = "0" validation

File: packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx [hunk 2]
```diff
@@ -6,15 +6,29 @@ import { BalanceInputProps } from "./types";
...
   value,
   placeholder = "0.0",
-  onChange,
+  onUserInput,
   currencyValue,
   inputProps,
   isWarning = false,
   ...props
 }) => {
+  const handleOnChange = (e: React.FormEvent<HTMLInputElement>) => {
```
Comment (hachiojidev) [inline packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx:15]: This should be `React.ChangeEventHandler<HTMLInputElement>`

**Review by hachiojidev — CHANGES_REQUESTED** [review]
@memoyil Since this is a breaking change when this is released can you update the exchange and main site?

File: packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx (hunk 2 above)
Comment (memoyil) [reply packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx:15]: I got errors when i changed to that one

Comment (memoyil) [reply packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx]: I think it is `React.ChangeEvent<HTMLInputElement>`

Comment (hachiojidev) [reply packages/pancake-uikit/src/components/BalanceInput/BalanceInput.tsx]: That looks right.

**Comment by memoyil** [comment]
@hachiojidev Done 

https://github.com/pancakeswap/pancake-frontend/pull/946

**Review by hachiojidev — APPROVED** [review]

**Comment by hachiojidev** [comment]
@RabbitDoge Can you check this again.

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**